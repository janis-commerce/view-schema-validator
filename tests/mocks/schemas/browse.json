{
    "service": "sac",
    "name": "claim-type-browse",
    "root": "Browse",
    "canExport": true,
    "canRefresh": false,
    "source": {
        "service": "sac",
        "namespace": "claim-type",
        "method": "browse",
        "resolve": false
    },
    "appearance": {
        "desktop": {
            "rowMinHeight": 70,
            "rowVerticalAlign": "top"
        }
    },
    "staticFilters": [
        {
            "name": "status",
            "target": "path",
            "value": {
                "static": "statusId"
            }
        },
        {
            "name": "status",
            "target": "path",
            "value": {
                "static": [
                    2,
                    4
                ]
            }
        }
    ],
    "themes": {
        "themeOne": {
            "new": "black",
            "closed": "green",
            "_default": "grey"
        },
        "themeTwo": {
            "new": "grey",
            "closed": "fizzgreen"
        }
    },
    "sortableFields": [
        {
            "name": "test"
        },
        {
            "name": "test1",
            "isDefaultSort": true
        },
        {
            "name": "test2",
            "initialSortDirection": "asc"
        }
    ],
    "filters": [
        {
            "name": "filterInput",
            "label": "test.test.test",
            "component": "Input"
        },
        {
            "name": "localSelect",
            "component": "Select",
            "componentAttributes": {
                "translateLabels": true,
                "canClear": true,
                "options": {
                    "scope": "local",
                    "values": [
                        {
                            "label": "test",
                            "value": 1
                        }
                    ]
                }
            }
        },
        {
            "name": "remoteSelect",
            "component": "Select",
            "componentAttributes": {
                "translateLabels": true,
                "options": {
                    "scope": "remote",
                    "endpoint": {
                        "service": "sac",
                        "namespace": "claim-type",
                        "method": "list",
                        "resolve": false
                    },
                    "endpointParameters": [
                        {
                            "name": "status",
                            "target": "path",
                            "value": {
                                "static": 2
                            }
                        },
                        {
                            "name": "status",
                            "target": "query",
                            "value": {
                                "static": 1
                            }
                        }
                    ],
                    "valuesMapper": {
                        "label": {
                            "template": "{0} {1} - ({2})",
                            "fields": [
                                "firstname",
                                "lastname",
                                "email"
                            ]
                        },
                        "value": "id"
                    }
                }
            }
        },
        {
            "name": "remoteMultiselect",
            "component": "Multiselect",
            "componentAttributes": {
                "translateLabels": true,
                "options": {
                    "scope": "remote",
                    "initialValuesEndpoint": {
                        "service": "sac",
                        "namespace": "claim-type",
                        "method": "list",
                        "resolve": false
                    },
                    "initialValuesFilterName": "id",
                    "endpoint": {
                        "service": "sac",
                        "namespace": "claim-type",
                        "method": "list",
                        "resolve": false
                    },
                    "valuesMapper": {
                        "label": "name",
                        "value": "id"
                    }
                }
            }
        },
        {
            "name": "dateTimePickerFilter",
            "component": "DateTimePicker",
            "componentAttributes": {
                "selectDate": false,
                "selectTime": true,
                "selectRange": true
            }
        }
    ],
    "fields": [
        {
            "name": "id",
            "component": "BoldText",
            "attributes": {
                "sortable": true,
                "isDefaultSort": true
            },
            "deviceDisplay": "desktop",
            "mapper": "addHashtag"
        },
        {
            "name": "motiveName",
            "component": "MediumText",
            "deviceDisplay": "mobile",
            "filter": {
                "component": "Input"
            }
        },
        {
            "name": "parentName",
            "component": "Text",
            "mapper": {
                "name": "suffix",
                "props": {
                    "value": ".test"
                }
            }
        },
        {
            "name": "exampleBadgeLetter",
            "component": "BadgeLetter"
        },
        {
            "name": "otherExampleBadgeLetter",
            "component": "BadgeLetter",
            "componentAttributes": {
                "translateLabels": false,
                "backgroundColorSource": "background",
                "fontColorSource": "colorName"
            }
        },
        {
            "name": "date",
            "component": "Text",
            "mapper": {
                "name": "date",
                "props": {
                    "format": "DD/MM/YYYY"
                }
            }
        },
        {
            "name": "nameTest",
            "component": "Text",
            "mapper": {
                "name": "prefix",
                "props": {
                    "value": "common.names."
                }
            }
        },
        {
            "name": "linkTest",
            "component": "Link"
        },
        {
            "name": "linkTest2",
            "component": "Link",
            "componentAttributes": {
                "translateLabels": true,
                "labelField": "label",
                "label": "test",
                "target": "_self"
            }
        },
        {
            "name": "user1",
            "component": "UserChip"
        },
        {
            "name": "user2",
            "component": "UserChip",
            "componentAttributes": {
                "userDataSource": {
                    "email": "email",
                    "firstname": "firstname",
                    "lastname": "lastname",
                    "image": "image"
                }
            }
        },
        {
            "name": "name",
            "component": "LightText",
            "filter": {
                "component": "Input"
            },
            "attributes": {
                "sortable": true
            }
        },
        {
            "name": "appliesToLogistics",
            "component": "Chip",
            "mapper": "booleanToWord",
            "filter": {
                "component": "Select",
                "componentAttributes": {
                    "translateLabels": true,
                    "labelPrefix": "common.boolean.",
                    "options": [
                        {
                            "label": "yes",
                            "value": 1
                        },
                        {
                            "label": "no",
                            "value": 0
                        }
                    ]
                }
            }
        },
        {
            "name": "appliesToLogisticsRemote",
            "component": "Chip",
            "mapper": "booleanToWord",
            "filter": {
                "component": "Select",
                "remote": true,
                "componentAttributes": {
                    "translateLabels": true,
                    "options": {
                        "endpoint": {
                            "service": "view",
                            "namespace": "menu",
                            "method": "list",
                            "resolve": false
                        },
                        "searchParam": "filters[name]",
                        "valuesMapper": {
                            "label": "name",
                            "value": "id"
                        }
                    }
                }
            }
        },
        {
            "name": "flags",
            "component": "Chip",
            "label": "sac.entities.claimType.fields.appliesTo",
            "mapper": "translate",
            "componentAttributes": {
                "icon": "icon_test"
            },
            "filter": {
                "component": "Select",
                "remote": true,
                "componentAttributes": {
                    "translateLabels": true
                }
            }
        },
        {
            "name": "areaInCharge",
            "component": "Chip"
        },
        {
            "name": "colorOne",
            "component": "Color"
        },
        {
            "name": "colorTwo",
            "component": "Color",
            "componentAttributes": {
                "showCode": true
            }
        },
        {
            "name": "sacClaimChange",
            "component": "SacClaimChange"
        },
        {
            "name": "sla",
            "component": "TimeChip",
            "mapper": {
                "name": "numberToTime",
                "props": {
                    "type": "hour"
                }
            }
        },
        {
            "name": "status",
            "component": "StatusChip",
            "attributes": {
                "isStatus": true,
                "sortable": true
            },
            "componentAttributes": {
                "useTheme": true
            },
            "mapper": "translate",
            "filter": {
                "component": "Select",
                "componentAttributes": {
                    "translateLabels": true,
                    "options": [
                        {
                            "label": "common.status.active",
                            "value": 1
                        },
                        {
                            "label": "common.status.inactive",
                            "value": 0
                        }
                    ]
                }
            }
        },
        {
            "name": "statusWithThemeCustom",
            "component": "StatusChip",
            "componentAttributes": {
                "useTheme": "themeOne"
            }
        },
        {
            "name": "actions",
            "component": "ActionButtons",
            "componentAttributes": {
                "actionsData": [
                    {
                        "name": "testAction",
                        "icon": "star_light",
                        "color": "fizzGreen",
                        "type": "endpoint",
                        "options": {
                            "endpoint": {
                                "service": "sac",
                                "namespace": "claim",
                                "method": "get",
                                "resolve": false
                            },
                            "endpointParameters": {
                                "id": "id"
                            }
                        }
                    },
                    {
                        "name": "testAction2",
                        "icon": "star_light",
                        "color": "fizzGreen",
                        "type": "endpoint",
                        "options": {
                            "endpoint": {
                                "service": "sac",
                                "namespace": "claim",
                                "method": "get",
                                "resolve": false
                            },
                            "endpointParameters": [
                                {
                                    "name": "status",
                                    "target": "path",
                                    "value": {
                                        "dynamic": "id"
                                    }
                                },
                                {
                                    "name": "status",
                                    "target": "query",
                                    "value": {
                                        "static": 1
                                    }
                                }
                            ]
                        }
                    },
                    {
                        "name": "new",
                        "icon": "star_light",
                        "color": "fizzGreen",
                        "type": "link",
                        "options": {
                            "path": "/sac/claim-type/new"
                        },
                        "callback": "removeRow"
                    }
                ]
            }
        },
        {
            "name": "userCreated",
            "component": "AsyncWrapper",
            "componentAttributes": {
                "source": {
                    "service": "id",
                    "namespace": "user",
                    "method": "list",
                    "resolve": false
                },
                "dataMapping": {
                    "firstname": "firstname",
                    "lastname": "lastname",
                    "email": "email"
                },
                "field": {
                    "name": "user",
                    "component": "UserChip",
                    "componentAttributes": {
                        "userDataSource": {
                            "email": "email",
                            "firstname": "firstname",
                            "lastname": "lastname",
                            "image": "image"
                        }
                    }
                }
            }
        },
        {
            "name": "asyncUser",
            "component": "AsyncWrapper",
            "componentAttributes": {
                "source": {
                    "service": "id",
                    "namespace": "user",
                    "method": "list",
                    "resolve": false
                },
                "dataMapping": {
                    "firstname": "userTest"
                },
                "field": {
                    "name": "userTest",
                    "component": "Text"
                }
            }
        },
        {
            "name": "multiValueWrapperExample",
            "component": "MultiValueWrapper",
            "componentAttributes": {
                "useDataField": true,
                "field": {
                    "name": "areaInCharge",
                    "component": "Chip"
                }
            }
        },
        {
            "name": "multiValueWrapperExampleTwo",
            "component": "MultiValueWrapper",
            "componentAttributes": {
                "direction": "horizontal",
                "field": {
                    "name": "areaInCharge",
                    "component": "Chip"
                }
            }
        },
        {
            "name": "exampleFieldWithConditions",
            "component": "Text",
            "conditions": {
                "showWhen": [
                    [
                        {
                            "name": "isNotEmpty",
                            "field": [
                                "test",
                                "tes2"
                            ]
                        },
                        {
                            "name": "isNotEqualTo",
                            "field": "name",
                            "referenceValueType": "static",
                            "referenceValue": null
                        }
                    ],
                    [
                        {
                            "name": "isEmpty",
                            "field": "someField"
                        },
                        {
                            "name": "isOneOf",
                            "field": "someField",
                            "referenceValue": [
                                "test1",
                                "test2"
                            ]
                        },
                        {
                            "name": "isNotDev"
                        }
                    ]
                ]
            }
        },
        {
            "name": "exampleFieldWithConditionsTwo",
            "component": "Text",
            "conditions": {
                "showWhen": [
                    [
                        {
                            "name": "isEqualTo",
                            "field": "user1",
                            "referenceValueType": "dynamic",
                            "referenceValue": "name"
                        },
                        {
                            "name": "isNotOneOf",
                            "field": "someField",
                            "referenceValue": [
                                "test1",
                                "test2"
                            ]
                        }
                    ],
                    [
                        {
                            "name": "isNotEqualTo",
                            "field": [
                                "test",
                                "name"
                            ],
                            "referenceValue": true
                        }
                    ]
                ]
            }
        }
    ]
}