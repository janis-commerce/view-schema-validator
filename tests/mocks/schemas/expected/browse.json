{
    "service": "sac",
    "name": "claim-type-browse",
    "root": "Browse",
    "source": {
        "service": "sac",
        "namespace": "claim-type",
        "method": "browse",
        "resolve": false
    },
    "staticFilters": [
        {
            "name": "status",
            "target": "path",
            "value": {
                "static": "statusId"
            }
        },
        {
            "name": "status",
            "target": "path",
            "value": {
                "static": [2, 4]
            }
        }
    ],
    "filters": [
        {
            "name": "filterInput",
            "label": "test.test.test",
            "component": "Input",
            "componentAttributes": {}
        },
        {
            "name": "localSelect",
            "component": "Select",
            "componentAttributes": {
                "translateLabels": true,
                "options": {
                    "scope": "local",
                    "values": [
                        {
                            "label": "test",
                            "value": 1
                        }
                    ]
                }
            }
        },
        {
            "name": "remoteSelect",
            "component": "Select",
            "componentAttributes": {
                "translateLabels": true,
                "options": {
                    "scope": "remote",
                    "searchParam": "term",
                    "endpoint": {
                        "service": "sac",
                        "namespace": "claim-type",
                        "method": "list",
                        "resolve": false
                    },
                    "valuesMapper": {
                        "label": "name",
                        "value": "id"
                    }
                }
            }
        },
        {
            "name": "dateTimePickerFilter",
            "component": "DateTimePicker",
            "componentAttributes": {
                "selectDate": false,
                "selectTime": true,
                "selectRange": true
            }
        }
    ],
    "fields": [
        {
            "name": "id",
            "component": "BoldText",
            "attributes": {
                "sortable": true,
                "isDefaultSort": true,
                "isStatus": false,
                "initialSortDirection": "desc"
            },
            "mapper": "addHashtag",
            "componentAttributes": {
                "fontWeight": "bold"
            }
        },
        {
            "name": "motiveName",
            "component": "MediumText",
            "filter": {
                "component": "Input",
                "type": "equal",
                "remote": false,
                "componentAttributes": {}
            },
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "fontWeight": "medium"
            }
        },
        {
            "name": "parentName",
            "component": "Text",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "fontWeight": "normal"
            }
        },
        {
            "name": "date",
            "component": "Text",
            "mapper": {
                "name": "date",
                "props": {
                    "format": "DD/MM/YYYY"
                }
            },
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "fontWeight": "normal"
            }
        },
        {
            "name": "nameTest",
            "component": "Text",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "mapper": {
                "name": "prefix",
                "props": {
                    "value": "common.names."
                }
            },
            "componentAttributes": {
                "fontWeight": "normal"
            }
        },
        {
            "name": "linkTest",
            "component": "Link",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {}
        },
        {
            "name": "linkTest2",
            "component": "Link",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "translateLabels": true,
                "labelField": "label",
                "label": "test",
                "target": "_self"
            }
        },
        {
            "name": "user1",
            "component": "UserChip",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {}
        },
        {
            "name": "user2",
            "component": "UserChip",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
    			"userDataSource": {
                    "email": "email",
                    "firstname": "firstname",
                    "lastname": "lastname",
                    "images": "images"
                }
            }
        },
        {
            "name": "name",
            "component": "LightText",
            "filter": {
                "component": "Input",
                "type": "equal",
                "remote": false,
                "componentAttributes": {}
            },
            "attributes": {
                "sortable": true,
                "isStatus": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "fontWeight": "light"
            }
        },
        {
            "name": "appliesToLogistics",
            "component": "Chip",
            "mapper": "booleanToWord",
            "filter": {
                "component": "Select",
                "componentAttributes": {
                    "translateLabels": true,
                    "labelPrefix": "common.boolean.",
                    "options": [
                        {
                            "label": "yes",
                            "value": 1
                        },
                        {
                            "label": "no",
                            "value": 0
                        }
                    ]
                },
                "type": "equal",
                "remote": false
            },
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "borderColor": "grey"
            }
        },
        {
            "name": "appliesToLogisticsRemote",
            "component": "Chip",
            "mapper": "booleanToWord",
            "filter": {
                "component": "Select",
                "remote": true,
                "componentAttributes": {
                    "translateLabels": true,
                    "options": {
                        "endpoint": {
                            "service": "view",
                            "namespace": "menu",
                            "method": "list",
                            "resolve": false
                        },
                        "searchParam": "filters[name]",
                        "valuesMapper": {
                            "label": "name",
                            "value": "id"
                        }
                    }
                },
                "type": "equal"
            },
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "borderColor": "grey"
            }
        },
        {
            "name": "flags",
            "component": "Chip",
            "label": "sac.entities.claimType.fields.appliesTo",
            "mapper": "translate",
            "filter": {
                "component": "Select",
                "remote": true,
                "componentAttributes": {
                    "translateLabels": true
                },
                "type": "equal"
            },
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "icon": "icon_test",
                "borderColor": "grey"
            }
        },
        {
            "name": "areaInCharge",
            "component": "Chip",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "borderColor": "grey"
            }
        },
        {
            "name": "colorOne",
            "component": "Color",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {}
        },
        {
            "name": "colorTwo",
            "component": "Color",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "showCode": true
            }
        },
        {
            "name": "sacClaimChange",
            "component": "SacClaimChange",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {}
        },
        {
            "name": "sla",
            "component": "TimeChip",
            "mapper": {
                "name": "numberToTime",
                "props": {
                    "type": "hour"
                }
            },
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {}
        },
        {
            "name": "status",
            "component": "StatusChip",
            "attributes": {
                "isStatus": true,
                "sortable": true,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
    			"useTheme": true
            },
            "mapper": "translate",
            "filter": {
                "component": "Select",
                "componentAttributes": {
                    "translateLabels": true,
                    "options": [
                        {
                            "label": "common.status.active",
                            "value": 1
                        },
                        {
                            "label": "common.status.inactive",
                            "value": 0
                        }
                    ]
                },
                "type": "equal",
                "remote": false
            }
        },
        {
            "name": "actions",
            "component": "ActionButtons",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "actionsData": [
                    {
                        "name": "new",
                        "icon": "star_light",
                        "color": "fizzGreen",
                        "type": "link",
                        "options": {
                            "path": "/sac/claim-type/new"
                        },
                        "callback": "removeRow"
                    }
                ]
            }
        },
        {
            "name": "userCreated",
            "component": "AsyncWrapper",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "source": {
                    "service": "id",
                    "namespace": "user",
                    "method": "list",
                    "resolve": false
                },
                "dataMapping": {
                    "firstname": "firstname",
                    "lastname": "lastname",
                    "email": "email"
                },
                "field": {
                    "name": "user",
                    "component": "UserChip",
                    "componentAttributes": {
                        "userDataSource": {
                            "email": "email",
                            "firstname": "firstname",
                            "lastname": "lastname",
                            "images": "images"
                        }
                    },
                    "attributes": {
                        "isStatus": false,
                        "sortable": false,
                        "isDefaultSort": false,
                        "initialSortDirection": "desc"
                    }
                }
            }
        },
        {
            "name": "asyncUser",
            "component": "AsyncWrapper",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "source": {
                    "service": "id",
                    "namespace": "user",
                    "method": "list",
                    "resolve": false
                },
                "dataMapping": {
                    "firstname": "userTest"
                },
                "field": {
                    "name": "userTest",
                    "component": "Text",
                    "attributes": {
                        "isStatus": false,
                        "sortable": false,
                        "isDefaultSort": false,
                        "initialSortDirection": "desc"
                    },
                    "componentAttributes": {
                        "fontWeight": "normal"
                    }
                }
            }
        },
        {
            "name": "multiValueWrapperExample",
            "component": "MultiValueWrapper",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "direction": "vertical",
                "field": {
                    "name": "areaInCharge",
                    "component": "Chip",
                    "attributes": {
                        "isStatus": false,
                        "sortable": false,
                        "isDefaultSort": false,
                        "initialSortDirection": "desc"
                    },
                    "componentAttributes": {
                        "borderColor": "grey"
                    }
                }
            }
        },
        {
            "name": "multiValueWrapperExampleTwo",
            "component": "MultiValueWrapper",
            "attributes": {
                "isStatus": false,
                "sortable": false,
                "isDefaultSort": false,
                "initialSortDirection": "desc"
            },
            "componentAttributes": {
                "direction": "horizontal",
                "field": {
                    "name": "areaInCharge",
                    "component": "Chip",
                    "attributes": {
                        "isStatus": false,
                        "sortable": false,
                        "isDefaultSort": false,
                        "initialSortDirection": "desc"
                    },
                    "componentAttributes": {
                        "borderColor": "grey"
                    }
                }
            }
        }
    ],
    "hasPreview": false,
    "canEdit": true,
    "canCreate": true,
    "canView": false,
    "pageSize": 60,
    "actions": [
        {
            "name": "new",
            "icon": "star_light",
            "color": "fizzGreen",
            "type": "link",
            "options": {
                "path": "/sac/claim-type/new"
            }
        }
    ]
}